src/gluonts/model/wavenet/_network.py

```diff
@@ -568,40 +568,51 @@ def blow_up(u):
 
         res = F.slice_axis(past_target, begin=-2, end=None, axis=-1)
         res = blow_up(res)
+        scale_bu = blow_up(scale)
         for n in range(self.pred_length):
             # Generate one-step ahead predictions. The input consists of target and features
             # corresponding to the last two time steps.
             current_target = F.slice_axis(res, begin=-2, end=None, axis=-1)
+
+            input_tar_repr_loc, _, _ = self.input_repr(
+                current_target,
+                F.ones_like(current_target),
+                scale_bu,
+                rep_params_in,
+            )
+            input_tar_repr_loc = input_tar_repr_loc.swapaxes(1, 2)
+            _, _, rep_params = self.output_repr(
+                current_target,
+                F.ones_like(current_target),
+                scale_bu,
+                rep_params_out,
+            )
+
             current_features = F.slice_axis(
                 full_features,
                 begin=self.receptive_field + n - 1,
                 end=self.receptive_field + n + 1,
                 axis=-1,
             )
             embedding = self.target_feature_embedding(
-                F, target=current_target, features=blow_up(current_features),
+                F,
+                target=input_tar_repr_loc,
+                features=blow_up(current_features),
             )
 
             # (batch_size, 1, num_bins) where 1 corresponds to the time axis.
             unnormalized_outputs, queues = self.base_net(
                 F, embedding, one_step_prediction=True, queues=queues
             )
-            if self.temperature > 0:
-                # (batch_size, 1, num_bins) where 1 corresponds to the time axis.
-                probs = F.softmax(
-                    unnormalized_outputs / self.temperature, axis=-1
-                )
-                # (batch_size, 1)
-                y = F.sample_multinomial(probs)
-            else:
-                # (batch_size, 1)
-                y = F.argmax(unnormalized_outputs, axis=-1)
-            y = y.astype("int32")
+            distr_args = self.proj_distr_args(unnormalized_outputs)
+            distr = self.distr_output.distribution(distr_args, scale=scale_bu)
+
+            y = distr.sample()
```

Comment:
Earlier we had a way to sample only the mode if we set `temperature` to zero or negative. We don't have this now. Also it looks like `distr_output` does not take into account `temperature`.

Reply:
`CategoricalOutput()` includes a `temperature` parameter which you can set to your liking to regain that same functionality: https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/distribution/categorical.py#L106

Reply:
Yes, I saw that but you are not setting the `temperature` that is passed to `WavenetSampler`. At the moment passing a different value to `temperature` and using the default value for `distr_output` in `WavenetEstimator` cannot reproduce earlier behavior (`temperature` is silently ignored). 

I see that `temperature` is set inside `CategoricalOutput` and it becomes difficult to reset it given that both `temperature` and `distr_output` are constructor arguments of `WavenetEstimator`.  

This brings the question: should `temperature` be part of `CategoricalOutput`? Currently it is used to modify sampling alone. If there are no other cases, why not pass it as an explicit argument to `sample` method?